repositories.bzl

```diff
@@ -0,0 +1,559 @@
+def boringssl_repositories(bind=True):
+    native.git_repository(
+        name = "boringssl",
+        commit = "bfd36df3da38dbf8828e712f42fbab2a0034bc40",  # 2017-02-02
+        remote = "https://boringssl.googlesource.com/boringssl",
+    )
+
+    if bind:
+        native.bind(
+            name = "boringssl_crypto",
+            actual = "@boringssl//:crypto",
+        )
+
+        native.bind(
+            name = "libssl",
+            actual = "@boringssl//:ssl",
+        )
+
+
+def protobuf_repositories(bind=True):
+    native.git_repository(
+        name = "protobuf_git",
+        commit = "a428e42072765993ff674fda72863c9f1aa2d268",  # v3.1.0
+        remote = "https://github.com/google/protobuf.git",
+    )
+
+    if bind:
+        native.bind(
+            name = "protoc",
+            actual = "@protobuf_git//:protoc",
+        )
+
+        native.bind(
+            name = "protobuf",
+            actual = "@protobuf_git//:protobuf",
+        )
+
+        native.bind(
+            name = "cc_wkt_protos",
+            actual = "@protobuf_git//:cc_wkt_protos",
+        )
+
+        native.bind(
+            name = "cc_wkt_protos_genproto",
+            actual = "@protobuf_git//:cc_wkt_protos_genproto",
+        )
+
+        native.bind(
+            name = "protobuf_compiler",
+            actual = "@protobuf_git//:protoc_lib",
+        )
+
+def googletest_repositories(bind=True):
+    BUILD = """
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+#
+
+cc_library(
+    name = "googletest",
+    srcs = [
+        "googletest/src/gtest-all.cc",
+        "googlemock/src/gmock-all.cc",
```

Comment:
Buildifier: `googlemock/src/gmock-all.cc` should be listed first (alphabetical order).